### DISCUSSION TITLE
Pre-Clonotypes

### DISCUSSION THREAD

**Discussion by bshim181** [milaboratory/mixcr#1736] [Q&A] [answered]
Hello, 

I am running in to issues where there are large number of zero pre-clonotypes groups. I am looking at different replicates for the sample run and it varies from 65% to high 80 range. What might be the cause for this?

The reads have 12 bp UMI with around 500,000 reads in total. It demonstrates about 95% alignment rate. 

<img width="792" alt="Screenshot 2024-08-05 at 2 04 37 PM" src="https://github.com/user-attachments/assets/12bffdfa-623f-4b8b-af8e-510caf203e70">

**Comment by mizraelson** [comment, marked as answer]
This means that for the majority of UMI groups, there are multiple clones marked with the same UMI sequence.

**Reply by bshim181** [reply]
is there a way to pull out all sequences of single UMI group and see what CDR3 consensus sequences were present?

**Reply by mizraelson** [reply]
Yes, you can export the alignments from the VDJCA file, including barcodes. Then you can group the table by the UMI barcode sequence and check the number of unique CDR3s per group.

```
mixcr exportAlignments \
   -tags Molecule \
   inpute.refined.vdjca \
   alignments.tsv
```

**Reply by bshim181** [reply]
<img width="862" alt="Screenshot 2024-08-07 at 4 03 55 PM" src="https://github.com/user-attachments/assets/91e57a13-0c44-4c9e-bb13-82faf4b96cac">
<img width="958" alt="Screenshot 2024-08-07 at 4 02 07 PM" src="https://github.com/user-attachments/assets/033bb0d7-04da-4c0f-9995-6cd771cfb980">
<img width="946" alt="Screenshot 2024-08-07 at 4 02 38 PM" src="https://github.com/user-attachments/assets/08200d9c-e6a0-4552-aca8-6ff553ee13a9">

Hello, I have attached the alignment, refineTagsandSort, and assembly report. I was wondering if you could help with figuring out what might be causing the issue. We do not want to exclude the pre-cloning step as a whole because that just removes the whole point of utilizing UMIs.

**Reply by mizraelson** [reply]
Hi, from the reports it is only clear that a lot of data is lost during pre-clonotype assembly step. To investigate you will need to look into the alignments. using the command above. Did you see multiple CDR3 per UMI there?

**Reply by bshim181** [reply]
Yes. from the alignments.tsv file, I grouped together reads with same UMI and then compared their CDR3. I found several UMIs with very frequent occurrences (more than 10,000). This is just example of UMI with around 100 reads corresponding to it. Below is the diversity of CDR3 within the same UMI. From what I see, I definitely do not see a level of consensus here. 
<img width="354" alt="Screenshot 2024-08-08 at 9 40 18 AM" src="https://github.com/user-attachments/assets/a352ea36-1a19-46b4-9f9f-e1e6e5a1ec4a">

**Reply by mizraelson** [reply]
Yes, so the question is what do you wanna do with it. Theoretically you can tune the parameters to extract more consensuses from a UMI group by adding:

-Massemble.consensusAssemblerParameters.assembler.maxConsensuses=10

Or even more then 10. There are also other parameters that you can tune to yield more data from each UMI group but most likely you will be still losing part of the data.
